src/gtest/test_arith_uint256.cpp

```diff
@@ -0,0 +1,567 @@
+    for (unsigned int i = 0; i < 256; ++i)
+    {
+        EXPECT_TRUE(ZeroL != (OneL << i));
+        EXPECT_TRUE((OneL << i) != ZeroL);
+        EXPECT_TRUE(R1L != (R1L ^ (OneL << i)));
+        EXPECT_TRUE(((arith_uint256(Tmp64) ^ (OneL << i) ) != Tmp64 ));
+    }
+    EXPECT_TRUE(ZeroL == (OneL << 256));
+
+    // String Constructor and Copy Constructor
+    EXPECT_TRUE(arith_uint256("0x"+R1L.ToString()) == R1L);
+    EXPECT_TRUE(arith_uint256("0x"+R2L.ToString()) == R2L);
+    EXPECT_TRUE(arith_uint256("0x"+ZeroL.ToString()) == ZeroL);
+    EXPECT_TRUE(arith_uint256("0x"+OneL.ToString()) == OneL);
+    EXPECT_TRUE(arith_uint256("0x"+MaxL.ToString()) == MaxL);
+    EXPECT_TRUE(arith_uint256(R1L.ToString()) == R1L);
+    EXPECT_TRUE(arith_uint256("   0x"+R1L.ToString()+"   ") == R1L);
+    EXPECT_TRUE(arith_uint256("") == ZeroL);
+    EXPECT_TRUE(R1L == arith_uint256(R1ArrayHex));
+    EXPECT_TRUE(arith_uint256(R1L) == R1L);
+    EXPECT_TRUE((arith_uint256(R1L^R2L)^R2L) == R1L);
+    EXPECT_TRUE(arith_uint256(ZeroL) == ZeroL);
+    EXPECT_TRUE(arith_uint256(OneL) == OneL);
+
+    // uint64_t constructor
+    EXPECT_TRUE( (R1L & arith_uint256("0xffffffffffffffff")) == arith_uint256(R1LLow64));
+    EXPECT_TRUE(ZeroL == arith_uint256(0));
+    EXPECT_TRUE(OneL == arith_uint256(1));
+    EXPECT_TRUE((arith_uint256("0xffffffffffffffff") = arith_uint256(0xffffffffffffffffULL)) == arith_uint256(0xffffffffffffffffULL) );
+
+    // Assignment (from base_uint)
+    arith_uint256 tmpL = ~ZeroL; EXPECT_TRUE(tmpL == ~ZeroL);
+    tmpL = ~OneL; EXPECT_TRUE(tmpL == ~OneL);
+    tmpL = ~R1L; EXPECT_TRUE(tmpL == ~R1L);
+    tmpL = ~R2L; EXPECT_TRUE(tmpL == ~R2L);
+    tmpL = ~MaxL; EXPECT_TRUE(tmpL == ~MaxL);
+
+}
+
+void shiftArrayRight(unsigned char* to, const unsigned char* from, unsigned int arrayLength, unsigned int bitsToShift)
+{
+    for (unsigned int T=0; T < arrayLength; ++T)
+    {
+        unsigned int F = (T+bitsToShift/8);
+        if (F < arrayLength)
+            to[T]  = from[F] >> (bitsToShift%8);
+        else
+            to[T] = 0;
+        if (F + 1 < arrayLength)
+            to[T] |= from[(F+1)] << (8-bitsToShift%8);
+    }
+}
+
+void shiftArrayLeft(unsigned char* to, const unsigned char* from, unsigned int arrayLength, unsigned int bitsToShift)
+{
+    for (unsigned int T=0; T < arrayLength; ++T)
+    {
+        if (T >= bitsToShift/8)
+        {
+            unsigned int F = T-bitsToShift/8;
+            to[T]  = from[F] << (bitsToShift%8);
+            if (T >= bitsToShift/8+1)
+                to[T] |= from[F-1] >> (8-bitsToShift%8);
+        }
+        else {
+            to[T] = 0;
+        }
+    }
+}
+
+TEST(arith_uint256_tests, shifts) // "<<"  ">>"  "<<="  ">>="
+{
+    unsigned char TmpArray[32];
+    arith_uint256 TmpL;
+    for (unsigned int i = 0; i < 256; ++i)
+    {
+        shiftArrayLeft(TmpArray, OneArray, 32, i);
+        EXPECT_TRUE(arith_uint256V(std::vector<unsigned char>(TmpArray,TmpArray+32)) == (OneL << i));
```

Comment:
add #include <vector_types.h> and replace all vector<unsigned char> with v_uint8.
all std:: can be removed in this file - you have "using namespace std" defined